File: backend/src/main/java/com/ufc/backend/backend/repositories/UserRepository.java
package com.ufc.backend.backend.repositories;

import com.ufc.backend.backend.model.Student;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.transaction.annotation.Transactional;


public interface UserRepository extends MongoRepository<Student, String> {
    @Transactional
    Student findByEmail(String email);
}

File: backend/src/main/java/com/ufc/backend/backend/repositories/UserRepository.kt
package com.ufc.backend.backend.repositories

import com.ufc.backend.backend.model.Student
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.transaction.annotation.Transactional

interface UserRepository : JpaRepository<Student, String> {
    @Transactional
    fun findByPersonEmail(email: String): Student?
}